openmetadata-service/src/main/java/org/openmetadata/service/search/indexes/TableIndex.java

```diff
@@ -46,22 +49,32 @@ public Map<String, Object> buildESDoc() {
     List<SearchSuggest> schemaSuggest = new ArrayList<>();
     List<SearchSuggest> databaseSuggest = new ArrayList<>();
     List<SearchSuggest> serviceSuggest = new ArrayList<>();
+    Set<List<TagLabel>> tagsWithChildren = new HashSet<>();
+    List<String> columnsWithChildrenName = new ArrayList<>();
     SearchIndexUtils.removeNonIndexableFields(doc, excludeFields);
     if (table.getColumns() != null) {
       List<FlattenColumn> cols = new ArrayList<>();
       parseColumns(table.getColumns(), cols, null);
 
       for (FlattenColumn col : cols) {
         columnSuggest.add(SearchSuggest.builder().input(col.getName()).weight(5).build());
+        columnsWithChildrenName.add(col.getName());
+        if (col.getTags() != null) {
+          tagsWithChildren.add(col.getTags());
+        }
       }
+      doc.put("columnNames", columnsWithChildrenName);
     }
     parseTableSuggest(suggest);
     serviceSuggest.add(SearchSuggest.builder().input(table.getService().getName()).weight(5).build());
     databaseSuggest.add(SearchSuggest.builder().input(table.getDatabase().getName()).weight(5).build());
     schemaSuggest.add(SearchSuggest.builder().input(table.getDatabaseSchema().getName()).weight(5).build());
     ParseTags parseTags = new ParseTags(Entity.getEntityTags(Entity.TABLE, table));
+    tagsWithChildren.add(parseTags.getTags());
+    List<TagLabel> flattenedTagList =
```

Comment:
this should be a set. Otherwise we will get duplicates

Reply:
above we are having it as a set of List<TagLabel> so it will be unique only right .. i have tested it adding duplicate values it works fine..